odoo/models.py

```diff
@@ -2643,6 +2643,9 @@ def _add_sql_constraints(self):
         foreign_key_re = re.compile(r'\s*foreign\s+key\b.*', re.I)
 
         for (key, definition, message) in self._sql_constraints:
+            if not definition:
+                # virtual constraint (e.g. implemented by a custom index)
+                continue
```

Comment:
Shouldn't we ensure the constraint exists?

Reply:
Yes, I changed it.
I'm not sure if it should be done in `self.pool.post_init` or `self.pool.post_constraint` though